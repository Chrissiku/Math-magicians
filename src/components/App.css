@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700;900&display=swap");

*,
*::before,
*::after {
  box-sizing: border-box;
  text-decoration: none;
  list-style: none;
  font-family: "Roboto", sans-serif;
}

body {
  margin: 0;
  background: rgb(196, 191, 191);
}

/* styling header */
header.header {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  padding: 2px 50px;
  border: 1px solid red;
  width: 100%;
}

header.header .logo {
  color: #000;
  font-size: 35px;
  font-weight: 900;
  line-height: 40px;
}

header.header nav ul {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 30px;
}

header.header nav ul li {
  background: rgb(173, 167, 167);
  text-align: center;
  padding: 10px 20px;
  border-radius: 30px;
  transition: all 0.4s;
}

header.header nav ul li:hover {
  background: rgb(105, 103, 103);
}

header.header nav ul li:hover > a {
  color: rgba(243, 134, 32, 0.925);
}

header.header nav ul li a {
  color: #000;
  font-size: 20px;
  line-height: 25px;
  font-weight: 500;
}

section {
  display: flex;
  flex-direction: column;
  padding: 2px 50px;
}

.quote{
 text-align: center;
}

section h1{
  font-size: 40px;
  color: rgb(214, 107, 7);
}

section .text{
  font-size: 20px;
  line-height: 30px;
}

.calculator-grid-container {
  display: grid;
  margin-top: 3rem;
  justify-content: center;
  grid-template-columns: repeat(4, 4rem);
  grid-template-rows: minmax(4rem, auto) repeat(6, 3rem);
}

.calculator-grid-container > button {
  cursor: pointer;
  font-size: 1rem;
  border: 1px solid rgb(197, 192, 192);
  outline: none;
  background-color: rgba(255, 255, 255, 0.9);
}

.calculator-grid-container > button.orange-color {
  background-color: rgba(255, 123, 0, 0.877);
}

.calculator-grid-container > button:hover,
.calculator-grid-container > button:focus {
  background-color: rgba(241, 193, 128, 0.9);
}

.span-two {
  grid-column: span 2;
}

.result {
  grid-column: 1 / -1;
  background-color: rgba(179, 174, 174, 0.925);
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: space-around;
  padding: 0.75rem;
  word-wrap: break-word;
  word-break: break-all;
}

.result .number {
  color: white;
  font-size: 2rem;
}
