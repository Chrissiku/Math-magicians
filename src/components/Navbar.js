import React from 'react';
import { Link } from 'react-router-dom';

const Navigation = () => (
  <header className="header">
    <Link className="logo" to="/">
      Math Magicians
    </Link>
    <nav>
      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/calculator">Calculator</Link>
        </li>
        <li>
          <Link to="/quote">Quote</Link>
        </li>
      </ul>
    </nav>
  </header>
);

export default Navigation;
