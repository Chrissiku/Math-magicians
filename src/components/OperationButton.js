/* eslint-disable react/prop-types */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

class OperationButton extends React.Component {
  render() {
    // eslint-disable-next-line react/prop-types
    return (
      <button className="orange-color" type="button">
        {this.props.operation}
      </button>
    );
  }
}

export default OperationButton;
