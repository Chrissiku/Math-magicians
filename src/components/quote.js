/* eslint-disable quotes */
import React from "react";

const Quote = () => (
  <>
    <section className="quote">
      <p className="text">
        A businessman is a hybrid of a dancer and a calculator. Paul Valery.
        <a href="https://www.brainyquote.com/topics/calculator-quotes">
          <br />
          Read More
        </a>
      </p>

      <p className="text">
        The mechanical loom and the calculator have shown us that technology is
        both disruptive and filled with opportunities. But it would be hard to
        find a decent argument that we would have been better off without these
        inventions. Oren Etzioni
        <a href="https://www.brainyquote.com/topics/calculator-quotes">
          <br />
          Read More
        </a>
      </p>

      <p className="text">
        People think that the arts are optional and they are not. They teach a
        level of emotional depth that is equally important to mathematic skill.
        You can replace some math skills with a calculator if you know how to
        operate the thing, but there is no calculator for human interaction. Hal
        Sparks
        <a href=" https://www.brainyquote.com/topics/calculator-quotes">
          <br />
          Read More
        </a>
      </p>

      <p className="text">
        My colleagues and I were engineers who worked for DCM&#39;s calculator
        division. These electronic calculators used digital integrated circuits,
        and then they started using chips. The advanced versions of those chips
        were used for programmable calculators, which were the forerunners of
        PCs. Shiv Nadar
        <a href=" https://www.brainyquote.com/topics/calculator-quotes">
          <br />
          Read More
        </a>
      </p>
    </section>
  </>
);

export default Quote;
