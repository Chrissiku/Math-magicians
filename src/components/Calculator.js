/* eslint-disable react/button-has-type */
import './App.css';
import React, { useState } from 'react';
import { Routes, Route } from 'react-router-dom';
import DigitButton from './component_mocks/DigitButton';
import FunctionButton from './component_mocks/FunctionButton';
import SpanTwo from './component_mocks/spanTwoButton';
import Display from './component_mocks/Display';
import OperationButton from './component_mocks/OperationButton';
import calculate from '../logic/calculate';
import Navigation from './Navbar';
import Home from './Home';
import Quote from './quote';

function Calculator() {
  const [data, setData] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const eHandler = (e) => {
    const val = e.target.textContent;
    setData(calculate(data, val));
  };

  return (
    <section>
      <h2>Let&#39;s do some math</h2>
      <div className="calculator-grid-container">
        <Display value={data} />
        {/* first row */}
        <FunctionButton e={eHandler} name="AC" />
        <FunctionButton e={eHandler} name="+/-" />
        <FunctionButton e={eHandler} name="%" />
        <OperationButton e={eHandler} name="&divide;" />
        {/* second row */}
        <DigitButton e={eHandler} name="7" />
        <DigitButton e={eHandler} name="8" />
        <DigitButton e={eHandler} name="9" />
        <OperationButton e={eHandler} name="x" />
        {/* third row */}
        <DigitButton e={eHandler} name="4" />
        <DigitButton e={eHandler} name="5" />
        <DigitButton e={eHandler} name="6" />

        <OperationButton e={eHandler} name="-" />
        {/* fourth row */}
        <DigitButton e={eHandler} name="1" />
        <DigitButton e={eHandler} name="2" />
        <DigitButton e={eHandler} name="3" />

        <OperationButton e={eHandler} name="+" />
        {/* fith row */}
        <SpanTwo e={eHandler} name="0" />
        <button onClick={eHandler} name=".">
          .
        </button>
        <OperationButton e={eHandler} name="=" />
      </div>
    </section>
  );
}

const Magicians = () => (
  <>
    <Navigation />
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="calculator" element={<Calculator />} />
      <Route path="quote" element={<Quote />} />
    </Routes>
  </>
);

export default Magicians;
