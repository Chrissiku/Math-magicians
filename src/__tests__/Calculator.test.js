import { Calculator } from "../components/Calculator";
import renderer from 'react-test-renderer';

describe("Testing Calculator", () => {
    it("Testing component with snapshot", () => {
        const DOMTREE = renderer.create(<Calculator />).toJSON();

        expect(DOMTREE).toMatchSnapshot();
    });
});