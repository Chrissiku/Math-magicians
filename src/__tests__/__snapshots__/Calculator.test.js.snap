// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator Testing component with snapshot 1`] = `
<section>
  <h2>
    Let's do some math
  </h2>
  <div
    className="calculator-grid-container"
  >
    <div
      className="result"
    >
      <div
        className="number"
      >
        
    0
      </div>
    </div>
    <button
      id="clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      id="clear"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      id="clear"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="span-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      name="."
      onClick={[Function]}
    >
      .
    </button>
    <button
      className="orange-color"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;
